src/server/game/Spells/Spell.cpp

```diff
@@ -6857,6 +6857,11 @@ void Spell::DoAllEffectOnLaunchTarget(TargetInfo& targetInfo, float* multiplier)
     if (!unit)
         return;
 
+    // Check for SPELL_ATTR7_INTERRUPT_ONLY_NONPLAYER and eventually cast the interrupt spell if the target is a non-player
```

Comment:
I disagree with where this code is placed - Silencing Shot has travel time and should interrupt on impact, not when started